app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java

```diff
@@ -0,0 +1,241 @@
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.view.MenuItemCompat;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.SearchView;
+import android.text.TextUtils;
+import android.util.SparseArray;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ListView;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.adapters.ScheduleSessionsListAdapter;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.views.TitlePickerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import timber.log.Timber;
+
+/**
+ * Created by duncanleo on 4/2/16.
+ */
+public class ScheduleFragment extends Fragment implements SearchView.OnQueryTextListener {
+    private final String SEARCH = "searchText";
+    private String searchText = "";
+    private SearchView searchView;
+    private boolean showBookmarkedOnly;
+
+    public static final String SCHEDULE_TAB_POSITION = "SCHEDULE_TAB_POSITION", SHOW_BOOKMARKED_ONLY = "SHOW_BOOKMARKED_ONLY";
+    private TitlePickerView scheduleTitlePicker;
+    private ViewPager scheduleViewPager;
+    private ScheduleViewPagerAdapter viewPagerAdapter;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+        OpenEventApp.getEventBus().register(this);
+        super.onCreate(savedInstanceState);
+
+        if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
+            searchText = savedInstanceState.getString(SEARCH);
+        }
+        if (getArguments() != null) {
+            showBookmarkedOnly = getArguments().getBoolean(SHOW_BOOKMARKED_ONLY, false);
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_schedule, container, false);
+        scheduleTitlePicker = (TitlePickerView) v.findViewById(R.id.schedule_title_picker);
+        scheduleViewPager = (ViewPager) v.findViewById(R.id.schedule_view_pager);
+        viewPagerAdapter = new ScheduleViewPagerAdapter(getChildFragmentManager());
+        scheduleViewPager.setAdapter(viewPagerAdapter);
+        scheduleTitlePicker.setupWithViewPager(scheduleViewPager);
+        return v;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle bundle) {
+        if (isAdded()) {
+            if (searchView != null) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/triplez/open-event-android/file/2406802117/issues/source?bid=3233850&fileBranchId=3233911#l79)